gateway/src/apicast/http_proxy.lua

```diff
@@ -81,6 +83,20 @@ local function absolute_url(uri)
     )
 end
 
+function fsize (path)
+  local file, err = open(path)
+
+  if err or not file then
+    return nil, err
+  end
+
+  local current = file:seek()      -- get current position
```

Comment:
This seems unnecessary as we already have a [file_reader](https://github.com/3scale/APIcast/blob/fdf5b2ef36adaf884795e12ed91aa9ce8ffba2f3/gateway/src/resty/file.lua#L13) function which is implemented with lua coroutines and will be much safer in my opinion.